packages/mui-material/src/Tabs/Tabs.js

```diff
@@ -443,10 +443,34 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
     scroll(scrollValue);
   };
 
+  const getFirstVisibleTab = (tabs) => {
+    const containerSize = vertical ? tabsRef.current.clientHeight : tabsRef.current[clientSize];
+    const containerStartBound = Math.round(
+      vertical ? tabsRef.current.scrollTop : tabsRef.current.scrollLeft,
+    );
+
+    const containerEndBound = Math.round(containerStartBound + containerSize);
+
+    return tabs.find((tab) => {
+      const centerPoint =
+        (vertical ? tab.offsetTop : tab.offsetLeft) +
+        (vertical ? tab.clientHeight : tab[clientSize]) / 2;
+      return centerPoint >= containerStartBound && centerPoint <= containerEndBound;
+    });
+  };
+
   const getScrollSize = () => {
     const containerSize = tabsRef.current[clientSize];
     let totalSize = 0;
     const children = Array.from(tabListRef.current.children);
+    const firstVisibleTab = getFirstVisibleTab(children);
+
+    if (
+      firstVisibleTab &&
+      (firstVisibleTab[clientSize] > containerSize || firstVisibleTab.clientHeight > containerSize)
```

Comment:
```suggestion
      (firstVisibleTab[clientSize] > containerSize || firstVisibleTab.clientHeight > containerSize)
```

For vertical you are checking the same thing twice. Should probably be double-checked. Note that `clientSize` is being set to `clientHeight` or `clientWidth` depending on the vertical prop.